### ISSUE TITLE
Map STL

### ISSUE THREAD

**Issue by Udayv56** [Udayv56/CPP#50]
#include <iostream>
#include<map>
#include<string>
using namespace std;

int main()
{   
    map<string , int> marks;
    marks["Uday"] = 98;
    marks["abhinav"] = 58;
    marks["vijay"] = 84;
    marks["hirday"] = 97;
    marks.insert({{"Hemant" , 69}, {"lavi", 85}});
    map<string , int> :: iterator it;
    for(it = marks.begin() ; it != marks.end() ; it++){
        cout<<(*it).first<<" "<<(*it).second<<endl;
    }
     cout<<"The size is: "<<marks.size()<<endl;
    cout<<"The max size is: "<<marks.max_size()<<endl;
    cout<<"The empty return value is: "<<marks.empty() <<endl;

  
    return 0;
}

**OUTPUT-**

Hemant 69
Uday 98
abhinav 58
hirday 97
lavi 85
vijay 84
The size is: 6
The max size is: 128102389400760775
The empty return value is: 0

Welcome! Today, we'll explore the `map` container from the Standard Template Library (STL). The `map` is an associative container that stores a collection of key-value pairs, where each key is unique. Let's analyze the code to understand how it works:

1. Creating and populating the map:
   - An instance of `map<string, int>` called `marks` is created. It associates strings (names) with integers (marks).
   - Elements are added to the `marks` map using the square bracket notation. For example, `marks["Uday"] = 98;` assigns the value 98 to the key "Uday".
   - The `insert` member function is used to add multiple elements to the `marks` map at once. Each element is specified as a pair of key-value using curly braces.

2. Iterating over the map:
   - An iterator `it` of type `map<string, int>::iterator` is initialized to the beginning of the `marks` map using `marks.begin()`.
   - A loop is used to iterate through the map, printing each key-value pair. The `first` member of the iterator represents the key, and the `second` member represents the value.

3. Map properties:
   - The `size` member function returns the number of elements in the map.
   - The `max_size` member function returns the maximum number of elements the map can hold.
   - The `empty` member function returns a boolean value indicating whether the map is empty or not.

In summary, the code demonstrates the usage of the `map` container from the STL. `map` is an associative container that stores key-value pairs. It provides efficient lookup, insertion, and deletion of elements based on the key. The code shows how to populate a map, iterate over its elements, and retrieve information such as the size, maximum size, and emptiness of the map.

I hope this explanation helps you understand the `map` container in the STL. If you have any further questions, feel free to ask!
